File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/[VB]-Official Visual Studio 2010 Samples for C# 4.0/ADO.NET Entity Framework/VB/Model/Phone.vb
' Copyright © Microsoft Corporation.  All Rights Reserved.
' This code released under the terms of the 
' Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)



Imports Microsoft.VisualBasic
Imports System.Collections.Generic

Namespace EmployeeTracker.Model

    ''' <summary>
    ''' Represents an employees phone number
    ''' </summary>
    Public Class Phone
        Inherits ContactDetail
        ''' <summary>
        ''' Usage values that are valid for phone numbers
        ''' </summary>
        Private Shared validUsageValues_Renamed() As String = { "Business", "Home", "Cell" }

        ''' <summary>
        ''' Gets a list of usage values that are valid for phone numbers
        ''' </summary>
        Public Overrides ReadOnly Property ValidUsageValues() As IEnumerable(Of String)
            Get
                Return validUsageValues_Renamed
            End Get
        End Property

        ''' <summary>
        ''' Gets or sets the actual phone number
        ''' </summary>
        Public Overridable Property Number() As String

        ''' <summary>
        ''' Gets or sets the extension associated with this phone number
        ''' </summary>
        Public Overridable Property Extension() As String
    End Class
End Namespace

File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/46872-Official Visual Studio 2010 Samples for C# 4.0/ADO.NET Entity Framework/C#/Model/Phone.cs
// Copyright © Microsoft Corporation.  All Rights Reserved.
// This code released under the terms of the 
// Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)


namespace EmployeeTracker.Model
{
    using System.Collections.Generic;

    /// <summary>
    /// Represents an employees phone number
    /// </summary>
    public class Phone : ContactDetail
    {
        /// <summary>
        /// Usage values that are valid for phone numbers
        /// </summary>
        private static string[] validUsageValues = new string[] { "Business", "Home", "Cell" };

        /// <summary>
        /// Gets a list of usage values that are valid for phone numbers
        /// </summary>
        public override IEnumerable<string> ValidUsageValues
        {
            get { return validUsageValues; }
        }

        /// <summary>
        /// Gets or sets the actual phone number
        /// </summary>
        public virtual string Number { get; set; }

        /// <summary>
        /// Gets or sets the extension associated with this phone number
        /// </summary>
        public virtual string Extension { get; set; }
    }
}
